File: apps/python/event_sourcing/command.py
"""Command base class and concrete commands."""

from abc import ABC
from typing import Union


class Command(ABC):
    """Base class for all commands."""

    pass


class RestockOrder(Command):
    """Command to request a restock order."""

    def __init__(self, quantity: int):
        self.quantity = quantity


# Type alias for all command types
COMMANDS = Union[RestockOrder, None]

File: apps/rust/src/command.rs
//! Command types and trait definition.

/// Base trait for all commands.
pub trait Command {}

/// Command to request a restock order.
#[derive(Debug, Clone, PartialEq)]
pub struct RestockOrder {
    pub quantity: i32,
}

impl RestockOrder {
    pub fn new(quantity: i32) -> Self {
        Self { quantity }
    }
}

impl Command for RestockOrder {}
